src/SourceGenerators/IntegrationToGenerate.cs

```diff
@@ -0,0 +1,25 @@
+// <copyright file="IntegrationToGenerate.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Immutable;
+
+namespace SourceGenerators;
+
+internal readonly record struct IntegrationToGenerate(string IntegrationType, ImmutableArray<TargetToGenerate> Targets)
+{
+    public readonly string IntegrationType = IntegrationType;
+    public readonly ImmutableArray<TargetToGenerate> Targets = Targets;
```

Comment:
Unfortunately `ImmutableArray<>` doesn't implement value equality - I suggest you use a wrapper type that tests for value equality, something like this:

https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace.SourceGenerators/Helpers/EquatableArray.cs